Add single support for numba_dpex_p

ZzEeKkAa opened the pull request:
Add support for single precision numba_dpex_p

- [x] Have you provided a meaningful PR description?
- [x] Have you added a test, reproducer or referred to an issue with a reproducer?
- [x] Have you tested your changes locally for CPU and GPU devices?
- [x] Have you made sure that new changes do not introduce compiler warnings?
- [ ] If this PR is a work in progress, are you filing the PR as a draft?

diptorupd commented:
@ZzEeKkAa the issue is the use of `dtype` function call inside the `prange`. It is not supported yet. The best I could do at the moment was to improve the error message: https://github.com/IntelPython/numba-dpex/pull/1061

I suggest you move all the `dtype` calls out of the `prange` loop body.

ZzEeKkAa commented:
Thank you @diptorupd. I've added issue in dpex to track this. https://github.com/IntelPython/numba-dpex/issues/1063. In a meantime I've added workarounds for it.
